addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -540,28 +542,61 @@ public void parseAndUpdate(String request, String payLoad, String result) {
-                            } else {
-                                if (!entry.getValue().isJsonNull()) {
-                                    Map<String, String> properties = editProperties();
-                                    properties.put(selector.getChannelID(), entry.getValue().getAsString());
-                                    updateProperties(properties);
+                        if (entry.getKey().equals("timestamp")) {
+                            resultTimeStamp = Long.valueOf(entry.getValue().getAsString());
+                            Date date = new Date();
+                            date.setTime(resultTimeStamp);
+                            logger.trace("The request result timestamp is {}", DATE_FORMATTER.format(date));
+                            break;
+                        }
+                    }
+
+                    try {
+                        lock.lock();
+
+                        boolean proceed = true;
+                        if (resultTimeStamp < lastTimeStamp && request == TESLA_DRIVE_STATE) {
+                            proceed = false;
+                        } else {
+                            lastTimeStamp = resultTimeStamp;
+                        }
+
+                        if (proceed) {
+                            for (Map.Entry<String, JsonElement> entry : entrySet) {
+                                try {
+                                    TeslaChannelSelector selector = TeslaChannelSelector
+                                            .getValueSelectorFromRESTID(entry.getKey());
+                                    if (!selector.isProperty()) {
+                                        if (!entry.getValue().isJsonNull()) {
+                                            updateState(selector.getChannelID(), teslaChannelSelectorProxy.getState(
+                                                    entry.getValue().getAsString(), selector, editProperties()));
+                                        } else {
+                                            updateState(selector.getChannelID(), UnDefType.UNDEF);
+                                        }
+                                    } else {
+                                        if (!entry.getValue().isJsonNull()) {
+                                            Map<String, String> properties = editProperties();
+                                            properties.put(selector.getChannelID(), entry.getValue().getAsString());
+                                            updateProperties(properties);
+                                        }
+                                    }
+                                } catch (IllegalArgumentException e) {
+                                    logger.trace("The variable/value pair '{}':'{}' is not (yet) supported",
+                                            entry.getKey(), entry.getValue());
+                                } catch (ClassCastException | IllegalStateException e) {
```

Comment:
It is thrown by JsonElement in fact, and since the argument for getAsTring() is the result of getValue(), you do not know in advance if the deserialised object is of the right type.